mitosheet/mitosheet/updates/existing_imports.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from copy import copy
+from typing import Any, List
+from mitosheet.step import Step
+from mitosheet.types import StepsManagerType
+from mitosheet.step_performers import EVENT_TYPE_TO_STEP_PERFORMER
+from mitosheet.step_performers.dataframe_import import DataframeImportStepPerformer
+from mitosheet.step_performers.import_steps.excel_import import ExcelImportStepPerformer
+from mitosheet.step_performers.import_steps.simple_import import SimpleImportStepPerformer
+from mitosheet.utils import create_step_id
+
+
+EXISTING_IMPORTS_UPDATE_EVENT = 'existing_import_update'
+EXISTING_IMPORTS_PARAMS = ['updated_import_objs']
+
+
+def execute_existing_imports_update(steps_manager: StepsManagerType, updated_import_objs: List) -> None:
+    """
+    Updates the step list with new import steps
+
+    We must preserve the order of the sheets within the import. Consider this example:
```

Comment:
Per comment above, we def should refactor with new types to avoid having to do all this nasty counting of how many times we update a step!